client/state/reducer-registry/test/index.js

```diff
@@ -0,0 +1,391 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import deepFreeze from 'deep-freeze';
+import { createStore } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import { DESERIALIZE, SERIALIZE } from 'state/action-types';
+import { createReducer, withSchemaValidation, isValidStateWithSchema } from 'state/utils';
+import { addReducers, combineReducersAndAddLater } from 'state/reducer-registry';
+jest.mock( 'lib/warn', () => () => {} );
+
+describe( 'reducer-registry', () => {
+	const load = { type: DESERIALIZE };
+	const write = { type: SERIALIZE };
+	const grow = { type: 'GROW' };
+	const schema = {
+		type: 'number',
+		minimum: 0,
+	};
+
+	const age = ( state = 0, action ) => ( 'GROW' === action.type ? state + 1 : state );
+	age.schema = schema;
+
+	const height = ( state = 160, action ) => ( 'GROW' === action.type ? state + 1 : state );
+	const count = ( state = 1, action ) => ( 'GROW' === action.type ? state + 1 : state );
+
+	const date = ( state = new Date( 0 ), action ) => {
+		switch ( action.type ) {
+			case 'GROW':
+				return new Date( state.getTime() + 1 );
+			case SERIALIZE:
+				return state.getTime();
+			case DESERIALIZE:
+				if ( isValidStateWithSchema( state, schema ) ) {
+					return new Date( state );
+				}
+				return new Date( 0 );
+			default:
+				return state;
+		}
+	};
+	date.hasCustomPersistence = true;
+
+	describe( '#combineReducersAndAddLater', () => {
+		let reducers;
+
+		beforeEach( () => {
+			reducers = combineReducersAndAddLater(
+				{
+					age,
+					height,
+				},
+				'test',
+				Object.create( null )
```

Comment:
Any reason we need `Object.create( null )` vs `{}` ?

Reply:
The object here is used as a directory. Thus, there is no need for it to inherit the built-in `Object` object with access to methods on `Object.prototype`.